core/src/plugins/filed/python/percona-xtrabackup/bareos-fd-percona-xtrabackup.py

```diff
@@ -522,5 +564,56 @@ def restore_object_data(self, ROP):
             )
         return bRC_OK
 
+    def end_dumper(self):
+        """
+        Wait for bareos_xtrabackup_dumper to terminate
+        """
+        bareos_xtrabackup_dumper_returncode = None
+        # Wait timeout before sending TERM to the process
+        timeout = 30
+        start_time = int(time.time())
+        sent_sigterm = False
+        while self.stream.poll() is None:
+            if int(time.time()) - start_time > timeout:
+                DebugMessage(
+                    100,
+                    "Timeout wait for bareos_xtrabackup_dumper PID %s to terminate\n"
+                    % (self.stream.pid),
+                )
+                if not sent_sigterm:
+                    DebugMessage(
+                        100,
+                        "sending SIGTERM to bareos_xtrabackup_dumper PID %s\n"
+                        % (self.stream.pid),
+                    )
+                    os.kill(self.stream.pid, signal.SIGTERM)
+                    sent_sigterm = True
+                    # Wait timeout after sending TERM to the process
+                    timeout = 30
```

Comment:
```suggestion
```
This line can be deleted, because the timeout variable is never altered anywhere.

Reply:
That is +30s timeout after sending SIGTERM to the subprocess. Used in line 576.

Reply:
> That is +30s timeout after sending SIGTERM to the subprocess. Used in line 576.

Sure, but that's done by `start_time = int(time.time())`, the value of the timeout variable is always 30, as that is assigned before the while loop. Assigning the same value again is unnecessary and can be confusing when reading the code. It would only make sense if you would assign a different value than 30, for example if you want a longer or shorter timeout than 30s to wait after sending SIGTERM.

Reply:
Is that ok if I put the value of 10s there?